src/codegen/llvm/LLVMIRBuilder.cpp

```diff
@@ -1301,6 +1306,42 @@ LLVMIRBuilder::createSizeCall(ExpressionNode *expr) {
     return value_;
 }
 
+Value *
+LLVMIRBuilder::createMaxMinCall(ExpressionNode *actual, bool isMax) {
+    auto decl = dynamic_cast<QualifiedExpression *>(actual)->dereference();
+    auto type = dynamic_cast<TypeDeclarationNode *>(decl)->getType();
+    if (type->isReal()) {
+        if (type->getSize() == 4) {
+            value_ = ConstantFP::getInfinity(builder_.getFloatTy(), isMax);
+        } else {
+            value_ = ConstantFP::getInfinity(builder_.getDoubleTy(), isMax);
+        }
+    } else if (type->isInteger()) {
+        if (type->getSize() == 8) {
+            if (isMax) {
+                value_ = builder_.getInt64((uint64_t)(LLONG_MAX));
```

Comment:
Wouldn't it be better to use `std::numeric_limits<long>::max()` here?

Reply:
Yes. Lets avoid C macros. Note that `long` has different size on Linux & Windows 64bit.
I used `int64_t `... here to avoid this problem.